### PR TITLE
Introduce EmbeddedXlTable.defaults to store composite tag info

### PR DESCRIPTION
**Pull request by olejandro**
This PR introduces EmbeddedXlTable.defaults to store the additional info from a composite tag. Separating this info early on allows removing repeated decomposition of the composite tag.

### CODE REVIEW

File: xl2times/transforms.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def remove_comment_rows(
     result = []
 
     for table in tables:
-        tag = datatypes.Tag(table.tag.split(":")[0])
+        tag = datatypes.Tag(table.tag)
```
Comment (SamRWest) [inline xl2times/transforms.py:60]: Random nitpickey suggestion : this file would be more readable if we did `from .datatypes import Tag` and changed all these to just `Tag(table.tag)`. 
Unless there's a good reason not to (like if there was another Tag class somewhere that needed distinguishing from)

Same with `datatype.Config, EmbeddedXlTable and TimesModel` references.

**Review by SamRWest — COMMENTED** [review]
Can't comment on the logic (which I don't really understand), but other than my one nitpick, looks good :)

File: xl2times/transforms.py (hunk 1 above)
Comment (olejandro) [reply xl2times/transforms.py:60]: I think this would make a lot of sense. If @siddharth-krishna agrees we could change all them in a separate PR?

**Comment by olejandro** [comment]
> Can't comment on the logic (which I don't really understand), but other than my one nitpick, looks good :)

Thanks! Basically, this just extracts the extra info that may be included with a tag (e.g. `~FI_T: ACT_BND`) earlier. It will also make it easier to handle some other ways in which this extra info may be included.

File: xl2times/utils.py [hunk 2]
```diff
@@ -44,12 +44,14 @@ def apply_composite_tag(table: datatypes.EmbeddedXlTable) -> datatypes.EmbeddedX
     :return:           Table in EmbeddedXlTable format with declarations applied
                        and table tag simplified.
     """
-    if ":" in table.tag:
-        (newtag, varname) = table.tag.split(":")
-        varname = varname.strip()
+    if table.defaults:
+        varname = table.defaults
         df = table.dataframe.copy()
```
Comment (siddharth-krishna) [inline xl2times/utils.py:49]: I don't think we need the copy here btw

**Review by siddharth-krishna — APPROVED** [review]

File: xl2times/transforms.py (hunk 1 above)
Comment (siddharth-krishna) [reply xl2times/transforms.py:60]: Nice idea, thanks, I opened an issue to track it: https://github.com/etsap-TIMES/xl2times/issues/220

**Pull request merged**